machineconfiguration/v1/types.go

```diff
@@ -760,3 +760,137 @@ type ContainerRuntimeConfigList struct {
+	//              ...
+	//      ]
+	//
+	// These image references should all be by digest, tags aren't allowed.
+	// +optional
+	PinnedImages []PinnedImageRef `json:"pinnedImages,omitempty"`
+}
+
+// PinnedImageRef is an image reference by digest.
+//
+// +kubebuilder:validation:Pattern:=`@sha256:[a-fA-F0-9]{64}$`
+type PinnedImageRef string
+
+// PinnedImageSetStatus defines the observed state of a PinnedImageSet.
+type PinnedImageSetStatus struct {
+	// observedGeneration represents the generation observed by the controller.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+	
+	// conditions represents the latest available observations of current state.
+	// +listType=atomic
+	// +optional
+	Conditions []PinnedImageSetCondition `json:"conditions"`
+}
+
+// PinnedImageSetCondition contains condition information for a PinnedImageSet.
+type PinnedImageSetCondition struct {
+	// type specifies the state of pinned image set.
+	// +optional
+	Type PinnedImageSetConditionType `json:"type"`
+
+	// status of the condition, one of True, False, Unknown.
+	// +optional
+	Status corev1.ConditionStatus `json:"status"`
+
+	// lastTransitionTime is the time of the last update to the current status object.
+	// +optional
+	// +nullable
+	LastTransitionTime metav1.Time `json:"lastTransitionTime"`
+
+	// reason is the reason for the condition's last transition.
+	// +optional
+	Reason string `json:"reason,omitempty"`
+
+	// message provides additional information about the current condition. This is only to
+	// be consumed by humans.
+	// +optional
+	Message string `json:"message,omitempty"`
+}
+
+// PinnedImageSetConditionType valid conditions of a PinnedImageSet.
+type PinnedImageSetConditionType string
+
+const (
+	// PinnedImageSetControllerRunning means the pinned image set controller is running.
+	PinnedImageSetControllerRunning PinnedImageSetConditionType = "PinnedImageSetControllerRunning"
+
+	// PinnedImageSetControllerCompleted means the pinned image set controller has completed reconciliation.
+	PinnedImageSetControllerCompleted PinnedImageSetConditionType = "PinnedImageSetControllerCompleted"
+
+	// PinnedImageSetControllerFailing means the pinned image set controller controller is failing.
+	PinnedImageSetControllerFailing PinnedImageSetConditionType = "PinnedImageSetControllerFailing"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PinnedImageSetList is a list of PinnedImageSet resources
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+type PinnedImageSetList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []PinnedImageSet `json:"items"`
+}
+
+// NodeStatus provides information about the current state of a particular node managed by this operator.
+type NodeStatus struct {
```

Comment:
Curious, how are we planning to use this status in MCO? Perhaps an addition to @cdoern 's Status Reporting https://github.com/openshift/api/pull/1596.

Reply:
see below